src/Stratis.Bitcoin.Features.PoA/Voting/ReconstructFederationService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using NBitcoin;
+using NLog;
+
+namespace Stratis.Bitcoin.Features.PoA.Voting
+{
+    public sealed class ReconstructFederationService
+    {
+        private readonly IFederationManager federationManager;
+        private readonly IIdleFederationMembersKicker idleFederationMembersKicker;
+        private readonly Logger logger;
+        private readonly PoAConsensusOptions poaConsensusOptions;
+        private readonly VotingManager votingManager;
+
+        private bool isBusyReconstructing;
+
+        public ReconstructFederationService(
+            IFederationManager federationManager,
+            Network network,
+            IIdleFederationMembersKicker idleFederationMembersKicker,
+            VotingManager votingManager)
+        {
+            this.federationManager = federationManager;
+            this.idleFederationMembersKicker = idleFederationMembersKicker;
+            this.votingManager = votingManager;
+
+            this.logger = LogManager.GetCurrentClassLogger();
+            this.poaConsensusOptions = (PoAConsensusOptions)network.Consensus.Options;
+        }
+
+        public void Reconstruct(int height)
+        {
+            if (!this.poaConsensusOptions.VotingEnabled)
+            {
+                this.logger.Warn("Voting is not enabled on this node.");
+                return;
+            }
+
+            if (this.isBusyReconstructing)
```

Comment:
Is similarly named variable is being set elsewhere, but I don't see where this one is being set.